### ISSUE TITLE
Failure to build CRI-O on Fedora rawhide

### ISSUE THREAD

**Issue by adelton** [cri-o/cri-o#2680]
**Description**

Attempt to build CRI-O on Fedora rawhide fails with
```
(/cri-o/build/bin/go-md2man -in docs/crio.conf.5.md -out docs/crio.conf.5.tmp && touch docs/crio.conf.5.tmp && mv docs/crio.conf.5.tmp docs/crio.conf.5) || \
	(/cri-o/build/bin/go-md2man -in docs/crio.conf.5.md -out docs/crio.conf.5.tmp && touch docs/crio.conf.5.tmp && mv docs/crio.conf.5.tmp docs/crio.conf.5)
/bin/sh: /cri-o/build/bin/go-md2man: No such file or directory
/bin/sh: line 1: /cri-o/build/bin/go-md2man: No such file or directory
make: *** [Makefile:325: docs/crio.conf.5] Error 127
```

This is a regression against Fedora 30.

**Steps to reproduce the issue:**
1.
```
yum install -y \
  btrfs-progs-devel \
  containers-common \
  device-mapper-devel \
  git \
  glib2-devel \
  glibc-devel \
  glibc-static \
  go \
  gpgme-devel \
  libassuan-devel \
  libgpg-error-devel \
  libseccomp-devel \
  libselinux-devel \
  pkgconfig \
  runc
```
2. `git clone https://github.com/kubernetes-incubator/cri-o.git`
3. `dnf install -y make`
4. `( cd cri-o && make )`

**Describe the results you received:**
```
make: which: Command not found
mkdir -p "/cri-o/_output/src/github.com/cri-o"
ln -s "/cri-o" "/cri-o/_output/src/github.com/cri-o/cri-o"
touch "/cri-o/_output/.gopathok"
go build -ldflags '-s -w -X main.gitCommit="b94a84cd6d126b12c837db3650fed546376c1f58" -X main.buildInfo=1564398495' -tags "containers_image_ostree_stub       seccomp selinux" -o bin/crio github.com/cri-o/cri-o/cmd/crio
go build -ldflags '-s -w -X main.gitCommit="b94a84cd6d126b12c837db3650fed546376c1f58" -X main.buildInfo=1564398565' -tags "containers_image_ostree_stub       seccomp selinux" -o bin/crio-config github.com/cri-o/cri-o/cmd/crio-config
( cd conmon && /cri-o/bin/crio-config )
make -C conmon
make[1]: Entering directory '/cri-o/conmon'
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o ctr_logging.o ctr_logging.c
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o conmon.o conmon.c
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o cmsg.o cmsg.c
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o utils.o utils.c
cc -o ../bin/conmon config.h ctr_logging.o conmon.o cmsg.o utils.o -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0 -lglib-2.0  -lsystemd 
make[1]: Leaving directory '/cri-o/conmon'
make -C pause
make[1]: Entering directory '/cri-o/pause'
cc -std=c99 -Os -Wall -Wextra -static   -c -o pause.o pause.c
cc -o ../bin/pause pause.o -std=c99 -Os -Wall -Wextra -static 
make[1]: Leaving directory '/cri-o/pause'
./bin/crio --config="" config --default > crio.conf
go: github.com/golangci/golangci-lint@v1.17.1 requires
	github.com/go-critic/go-critic@v0.0.0-20181204210945-1df300866540: invalid pseudo-version: does not match version-control timestamp (2019-05-26T07:48:19Z)
(/cri-o/build/bin/go-md2man -in docs/crio.conf.5.md -out docs/crio.conf.5.tmp && touch docs/crio.conf.5.tmp && mv docs/crio.conf.5.tmp docs/crio.conf.5) || \
	(/cri-o/build/bin/go-md2man -in docs/crio.conf.5.md -out docs/crio.conf.5.tmp && touch docs/crio.conf.5.tmp && mv docs/crio.conf.5.tmp docs/crio.conf.5)
/bin/sh: /cri-o/build/bin/go-md2man: No such file or directory
/bin/sh: line 1: /cri-o/build/bin/go-md2man: No such file or directory
make: *** [Makefile:325: docs/crio.conf.5] Error 127
```

**Describe the results you expected:**
```
make: which: Command not found
mkdir -p "/cri-o/_output/src/github.com/cri-o"
ln -s "/cri-o" "/cri-o/_output/src/github.com/cri-o/cri-o"
touch "/cri-o/_output/.gopathok"
go build -ldflags '-s -w -X main.gitCommit="b94a84cd6d126b12c837db3650fed546376c1f58" -X main.buildInfo=1564398491' -tags "containers_image_ostree_stub       seccomp selinux" -o bin/crio github.com/cri-o/cri-o/cmd/crio
go build -ldflags '-s -w -X main.gitCommit="b94a84cd6d126b12c837db3650fed546376c1f58" -X main.buildInfo=1564398563' -tags "containers_image_ostree_stub       seccomp selinux" -o bin/crio-config github.com/cri-o/cri-o/cmd/crio-config
( cd conmon && /cri-o/bin/crio-config )
make -C conmon
make[1]: Entering directory '/cri-o/conmon'
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o ctr_logging.o ctr_logging.c
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o conmon.o conmon.c
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o cmsg.o cmsg.c
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0   -c -o utils.o utils.c
cc -o ../bin/conmon config.h ctr_logging.o conmon.o cmsg.o utils.o -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DVERSION=\"1.15.1-dev\" -DGIT_COMMIT=\""b94a84cd6d126b12c837db3650fed546376c1f58"\"  -D USE_JOURNALD=0 -lglib-2.0  -lsystemd 
make[1]: Leaving directory '/cri-o/conmon'
make -C pause
make[1]: Entering directory '/cri-o/pause'
cc -std=c99 -Os -Wall -Wextra -static   -c -o pause.o pause.c
cc -o ../bin/pause pause.o -std=c99 -Os -Wall -Wextra -static 
make[1]: Leaving directory '/cri-o/pause'
./bin/crio --config="" config --default > crio.conf
go: finding github.com/onsi/gomega v1.5.0
go: finding github.com/docker/docker v0.7.3-0.20190410184157-6d18c6a06295
go: finding github.com/cpuguy83/go-md2man v1.0.10
go: finding github.com/google/uuid v1.1.1
go: finding github.com/BurntSushi/toml v0.3.1
[...]
go: finding github.com/klauspost/compress v1.4.0
go: downloading github.com/cpuguy83/go-md2man v1.0.10
go: extracting github.com/cpuguy83/go-md2man v1.0.10
go: downloading github.com/russross/blackfriday v1.5.2
go: extracting github.com/russross/blackfriday v1.5.2
(/cri-o/build/bin/go-md2man -in docs/crio.conf.5.md -out docs/crio.conf.5.tmp && touch docs/crio.conf.5.tmp && mv docs/crio.conf.5.tmp docs/crio.conf.5) || \
	(/cri-o/build/bin/go-md2man -in docs/crio.conf.5.md -out docs/crio.conf.5.tmp && touch docs/crio.conf.5.tmp && mv docs/crio.conf.5.tmp docs/crio.conf.5)
(/cri-o/build/bin/go-md2man -in docs/crio.8.md -out docs/crio.8.tmp && touch docs/crio.8.tmp && mv docs/crio.8.tmp docs/crio.8) || \
	(/cri-o/build/bin/go-md2man -in docs/crio.8.md -out docs/crio.8.tmp && touch docs/crio.8.tmp && mv docs/crio.8.tmp docs/crio.8)
```

**Additional information you deem important (e.g. issue happens only occasionally):**
Installing `/usr/bin/go-md2man` via `golang-github-cpuguy83-md2man-2.0.0-0.3.20190624gitf79a8a8.fc31.x86_64` does not improve the situation.

**Output of `crio --version`:**
I don't have CRI-O built. But the repository is at
```
# ( cd cri-o && git describe --always )
b94a84cd6
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by saschagrunert** [comment]
Hey @adelton :wave: thanks for pointing to the issue. I tried to reproduce it within the fedora:latest container image, but it worked as intended. The cricital part seems to be:
```
go: github.com/golangci/golangci-lint@v1.17.1 requires
	github.com/go-critic/go-critic@v0.0.0-20181204210945-1df300866540: invalid pseudo-version: does not match version-control timestamp (2019-05-26T07:48:19Z)
```
We require an internet connection (go module discovery) when building the project outside of the `$GOPATH`. Could you give it another try with cloning the project into `$(go env GOPATH)/src/github.com/cri-o/cri-o` ? This should work, right?

**Comment by vrothberg** [comment]
@saschagrunert, does `go build -mode vendor` work?

**Comment by saschagrunert** [comment]
> @saschagrunert, does `go build -mode vendor` work?

I had the same idea so we might find a way to fix this. Nevertheless the `invalid pseudo-version` seems pretty strange to me. I'm wondering how this can be possible... :thinking:

**Comment by saschagrunert** [comment]
> > @saschagrunert, does `go build -mode vendor` work?
> 
> I had the same idea so we might find a way to fix this. :smile:  Nevertheless the `invalid pseudo-version` seems pretty strange to me. I'm wondering how this can be possible...

**Comment by vrothberg** [comment]
Awesome! IIRC, go 1.10 doesn't support `-mode vendor` yet, so I think we can pull the trigger when 1.10 isn't supported anymore.

**Comment by saschagrunert** [comment]
> Awesome! IIRC, go 1.10 doesn't support `-mode vendor` yet, so I think we can pull the trigger when 1.10 isn't supported anymore.

Yeah maybe we find a workaround for this as well. I'll prepare something for review...

**Comment by adelton** [comment]
> I tried to reproduce it within the fedora:latest container image, but it worked as intended.

Please try `registry.fedoraproject.org/fedora:rawhide` container image to reproduce.

**Comment by saschagrunert** [comment]
> > I tried to reproduce it within the fedora:latest container image, but it worked as intended.
> 
> Please try `registry.fedoraproject.org/fedora:rawhide` container image to reproduce.

Ah thanks, so the go issue is related to go 1.13. :) 

```
$ go version
go version go1.13beta1 linux/amd64
```

**Comment by saschagrunert** [comment]
Fix is incoming in #2682
